velox/exec/tests/SpillerAggregateInputBenchmarkTest.cpp

```diff
@@ -0,0 +1,159 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/exec/tests/SpillerAggregateInputBenchmarkTest.h"
+#include "velox/serializers/PrestoSerializer.h"
+
+#include <gflags/gflags.h>
+#include <deque>
+
+DEFINE_string(
+    spiller_benchmark_name,
+    "SpillerAggregateInputBenchmarkTest",
+    "The name of this benchmark");
+DEFINE_uint64(
+    spiller_benchmark_max_spill_file_size,
+    2 << 30,
+    "The max spill file size");
+DEFINE_uint64(
+    spiller_benchmark_write_buffer_size,
+    0,
+    "The spill write buffer size");
+DEFINE_uint64(
+    spiller_benchmark_min_spill_run_size,
+    1 << 30,
+    "The file directory path for spiller benchmark");
+DEFINE_uint32(
+    spiller_benchmark_num_spill_vectors,
+    1'000,
+    "The number of vectors for spilling");
+DEFINE_uint32(
+    spiller_benchmark_spill_vector_size,
+    500,
+    "The number of rows per each spill vector");
+DEFINE_string(
+    spiller_benchmark_compression_kind,
+    "none",
+    "The compression kind to compress spill rows before write to disk");
+DEFINE_uint32(
+    spiller_benchmark_spill_dependent_key_num,
+    2,
+    "The number of aggregation dependent key");
+
+using namespace facebook::velox;
+using namespace facebook::velox::common;
+using namespace facebook::velox::memory;
+using namespace facebook::velox::exec;
+
+namespace facebook::velox::exec::test {
+
+void SpillerAggregateInputBenchmarkTest::setUp() {
+  SpillerBenchmarkBase::setUp();
+
+  rowContainer_ = setupSpillContainer(
+      rowType_, FLAGS_spiller_benchmark_spill_dependent_key_num);
+  spiller_ = std::make_unique<Spiller>(
+      exec::Spiller::Type::kAggregateInput,
+      rowContainer_.get(),
+      [&](folly::Range<char**> rows) { rowContainer_->eraseRows(rows); },
+      rowType_,
+      HashBitRange{29, 29},
+      rowContainer_->keyTypes().size(),
+      std::vector<CompareFlags>{},
+      fmt::format("{}/{}", spillDir_, FLAGS_spiller_benchmark_name),
+      FLAGS_spiller_benchmark_max_spill_file_size,
+      FLAGS_spiller_benchmark_write_buffer_size,
+      FLAGS_spiller_benchmark_min_spill_run_size,
+      stringToCompressionKind(FLAGS_spiller_benchmark_compression_kind),
+      memory::spillMemoryPool(),
+      executor_.get());
+  spiller_->setPartitionsSpilled({0});
+  writeSpillData();
+}
+
+void SpillerAggregateInputBenchmarkTest::run() {
+  MicrosecondTimer timer(&executionTimeUs_);
+  spiller_->spill(0, 0);
+}
+
+void SpillerAggregateInputBenchmarkTest::writeSpillData() {
```

Comment:
```
s/SpillerAggregateInputBenchmarkTest/SpillerAggregateBenchmarkTest/
```
We can extend to support collecting stats on output? Also consider to override the print stats to log the detailed execution time? As for now, we only collect the time spent on aggregation input spill? Later, we can collect the time spent on output processing. We can skip collecting the time spent on writer spill data? Thanks!

Reply:
Got it. I used to think we needed to add two types of aggregate spill benchmarks, input type and output type. So I added `SpillerAggregateInputBenchmarkTest` to collect the time spent on aggregation input spill, and planned to add the output type of spill benchmark later.
Could I add `SpillerAggregateBenchmarkTest` with only input spill metrics in this PR, and extend it to support both types of spill benchmark and metrics collection in a follow-up PR?